### PR TITLE
feat: add buildFromOxlintConfigFile

### LINKED ISSUE #194: Support disabling all oxlint rules that are applied via the oxlint config file
This is a nice plugin, but I'm not sure it's aligned with the configuration file options. It would be great to have the plug-in detect/parse/use the config file so we don't have to explicitly keep things in sync in two places in order to transition to oxlint from eslint smoothly.

### PR DESCRIPTION
**Pull request by Sysix**
closes #194 

Long way for the goal:

- [x] support simple "rules" block
- [x] support simple "categories" block
- [x] support simple "plugins" block
- [x] support "categories" on, and "rules" off *
- [x] support "plugins" on, and "rules off *
- [x] numbers instead of `off`, `warn` or `error`
- [x] configuration array like: `no-magic_numbers: ["off", {...}]`
- [x] filter rules which are not enabled by plugin
- [x] all other wierd combinations
- [x] enable eslint always
- [x] check import from `../scripts` folder in build file
- [x] support jsonc / comments in files
- [x] prepare for overrides  https://github.com/oxc-project/oxc/pull/6974
- [x] Docs
- [x] more tests
  - [x] test rules count with oxlint command
  - [x] ~~depends on https://github.com/oxc-project/oxc/issues/6896~~ workaround in tests
  - [x] ~~depends on https://github.com/oxc-project/oxc/issues/7031~~ disabled `nursery` in integration test

\* the rules block overrides the rules from the other blocks, so this should be enabled in eslint

```json
{
  "categories": {
    "correctness": "error",
  }
  "rules": {
    "no-invalid-regexp": "off"
  }
}
```

### CODE REVIEW

**Comment by Sysix** [comment]
ready for feedback 🎉 . waiting for merge of #204 and #203

File: src/build-from-oxlint-config.spec.ts [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+          expect(rules.length).toBe(1);
+          expect(rules[0].rules).not.toBeUndefined();
+          expect('eqeqeq' in rules[0].rules!).toBe(true);
+          expect(rules[0].rules!.eqeqeq).toBe('off');
+        }
+      );
+    });
+
+    it('skip deactive rules inside "rules" scope', () => {
```
Comment (Sysix) [inline src/build-from-oxlint-config.spec.ts:30]: After sleeping 2 days. I do not know if I want to keep this behavior. 
When someone wants to disable it in oxlint, there is 2 reasons:

- they want it disabled in oxlint and eslint
- oxlint is not aligned with eslint

Comment (Boshen) [reply src/build-from-oxlint-config.spec.ts:30]: I don't have full context. Remove it if you don't feel it is correct.

Comment (Sysix) [reply src/build-from-oxlint-config.spec.ts:30]: When the config is:

```
{
  "rules": {
    "eqeqeq": "off",
  }
}
```

We do not disable the rule in eslint.

Comment (Boshen) [reply src/build-from-oxlint-config.spec.ts:30]: 🤔 

> We do not disable the rule in eslint.

This should be the correct behaviour?

Comment (Sysix) [reply src/build-from-oxlint-config.spec.ts:30]: That was I thought too.  But then I asked [why a user is disabling a rule](https://github.com/oxc-project/eslint-plugin-oxlint/pull/197#discussion_r1826442758).  
When the user do want the rule deactived in oxlint, why should it be actived in eslint?

Maybe this is only a small edge case and we can ignore it for the moment. ^^

**Pull request merged**